### PR TITLE
Add basic KUTTL tests for Cinder Operator

### PR DESCRIPTION
**Pull request by frenzyfriday**
This PR adds 2 very basic kuttl tests for cinder - to deploy and cleanup.

### CODE REVIEW

File: tests/kuttl/tests/keystone_scale/02-assert.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#
```
Comment (marios) [inline tests/kuttl/tests/keystone_scale/02-assert.yaml:1]: rename from keystone_scale

File: tests/kuttl/tests/keystone_scale/01-assert.yaml [hunk 2]
```diff
@@ -0,0 +1,244 @@
+#
```
Comment (cescgina) [inline tests/kuttl/tests/keystone_scale/01-assert.yaml:1]: If the first step just tests the deployment, this assert file should be just a symlink to `assert_sample_deployment`, to avoid duplication

File: tests/kuttl/tests/cinder_scale/01-assert.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+/home/anbanerj/Documents/next-gen/cinder-operator/tests/kuttl/common/assert_sample_deployment.yaml
```
Comment (cescgina) [inline tests/kuttl/tests/cinder_scale/01-assert.yaml:1]: This should be a relative path, i.e. `../../common/assert_sample_deployment.yaml

File: tests/kuttl/common/assert_sample_deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+              topologyKey: kubernetes.io/hostname
+            weight: 1
+      containers:
+      - args:
+        - -c
+        - /usr/local/bin/kolla_set_configs && /usr/local/bin/kolla_start
+        command:
+        - /bin/bash
+        env:
+        - name: CONFIG_HASH
```
Comment (cescgina) [inline tests/kuttl/common/assert_sample_deployment.yaml:77]: I'm not sure how this hash is generated but it probably changes between runs, will make the tests fail

File: tests/kuttl/common/assert_sample_deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+                  - cinder
+              topologyKey: kubernetes.io/hostname
+            weight: 1
+      containers:
+      - args:
+        - -c
+        - /usr/local/bin/kolla_set_configs && /usr/local/bin/kolla_start
+        command:
+        - /bin/bash
+        env:
```
Comment (cescgina) [inline tests/kuttl/common/assert_sample_deployment.yaml:76]: This complete `env` section should be removed, because if we remove just the hash, it fails because the env list have different lengths

File: tests/kuttl/common/assert_sample_deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+  cinderAPI:
+    replicas: 1
+    containerImage: quay.io/tripleozedcentos9/openstack-cinder-api:current-tripleo
+  cinderScheduler:
+    replicas: 1
+    containerImage: quay.io/tripleozedcentos9/openstack-cinder-scheduler:current-tripleo
+  cinderBackup:
+    replicas: 1
+    containerImage: quay.io/tripleozedcentos9/openstack-cinder-backup:current-tripleo
+  secret: cinder-secret
```
Comment (cescgina) [inline tests/kuttl/common/assert_sample_deployment.yaml:23]: In my environment this appears to be `osp-secret`, if it's something variable, it would be best to remove it

File: tests/kuttl/tests/keystone_scale/02-assert.yaml (hunk 1 above)
Comment (frenzyfriday) [reply tests/kuttl/tests/keystone_scale/02-assert.yaml]: done

File: tests/kuttl/tests/keystone_scale/01-assert.yaml (hunk 2 above)
Comment (frenzyfriday) [reply tests/kuttl/tests/keystone_scale/01-assert.yaml]: ack, done,  thanks

File: tests/kuttl/tests/cinder_scale/01-assert.yaml (hunk 3 above)
Comment (frenzyfriday) [reply tests/kuttl/tests/cinder_scale/01-assert.yaml]: ack, done,  thanks. Somehow it kept on taking the whole path from my local when I tried to create a symlink. I amended the multiple commits and merged the changes into one now.

File: tests/kuttl/common/assert_sample_deployment.yaml (hunk 4 above)
Comment (frenzyfriday) [reply tests/kuttl/common/assert_sample_deployment.yaml]: ack, removed

File: tests/kuttl/common/assert_sample_deployment.yaml (hunk 6 above)
Comment (frenzyfriday) [reply tests/kuttl/common/assert_sample_deployment.yaml:23]: ack, removing for now. In a separate PR I'll try to update this with something proper.

File: tests/kuttl/common/assert_sample_deployment.yaml (hunk 5 above)
Comment (frenzyfriday) [reply tests/kuttl/common/assert_sample_deployment.yaml]: ack, done,  thanks

**Comment by cescgina** [comment]
With some tweaks to the install_yaml targets, the tests are passing. I've noticed that we are just testing the `Cinder` CR, but there are others defined (`CinderAPI`, `CinderBackup`..., see https://github.com/openstack-k8s-operators/cinder-operator/tree/master/api/bases). We can either just provide some basic tests and let the dev team expand as needed or make them more complete here, both are valid solutions. At this point we should probably ask for review from the core team and get their feedback

**Review by abays — COMMENTED** [review]
+1 from me in regards to the actual basic KUTTL test

File: templates/cinder/config/cinder.conf [hunk 7]
```diff
@@ -95,12 +95,3 @@ lock_path = /var/locks/openstack/os-brick
 #volume_backend_name=nfs
 #volume_driver=cinder.volume.drivers.nfs.NfsDriver
 #nfs_shares_config=/etc/cinder/nfs_shares
-
```
Comment (marios) [inline templates/cinder/config/cinder.conf:98]: this and the changes in the README seem unrelated to kuttl tests though. This seems like a separate issue that was fixed (perhaps that you discovered working on the kuttl?).

Comment (frenzyfriday) [reply templates/cinder/config/cinder.conf:98]: I probably pulled this from somewhere trying to rebase, lemme check

Comment (frenzyfriday) [reply templates/cinder/config/cinder.conf]: I have rebased the PR, this should be gone now

**Review by abays — APPROVED** [review]
/lgtm
